client/gcs_service.go

```diff
@@ -23,7 +24,7 @@ type Service struct {
 type gcsStore interface {
 	WriteToObject(ctx context.Context, bucketName, fileName string, data []byte) error
 	ReadObject(ctx context.Context, bucketName, fileName string) ([]byte, error)
-	ListOjbect(ctx context.Context, bucketName, path string) ([]string, error)
+	ListObject(ctx context.Context, bucketName, path string) ([]google.SecretObject, error)
```

Comment:
It could be `google.Object` from google's package perspective (GCS object), it doesn't know anything about secrets. In callers, it could be used for secrets or users or any abstraction.